File: server/Models/Account.cs
namespace keepr.Models;

public class Creator
{
  public string Name { get; set; }
  public string Id { get; set; }
  public string Picture { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
}
public class Account : Creator
{
  public string Email { get; set; }
}

File: client/src/models/Account.js
export class Profile {
  constructor(data) {
    this.id = data.id
    this.name = data.name
    this.picture = data.picture
    // TODO add additional properties if needed
    this.coverImg = data.coverImg || 'https://images.unsplash.com/photo-1506744038136-46273834b3fb?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxzZWFyY2h8Mnx8Y292ZXJ8ZW58MHx8MHx8fDA%3D&w=1000&q=80'
  }
}
export class Account extends Profile {
  constructor(data) {
    super(data)
    this.email = data.email
  }
}

